src/prefect/agent/agent.py

```diff
@@ -673,7 +673,7 @@ def mark_failed(self, flow_run: GraphQLResult, exc: Exception) -> None:
             version=flow_run.version,
             state=Failed(message=str(exc)),
         )
-        self.logger.error("Error while deploying flow: {}".format(repr(exc)))
+        self.logger.error("Error while deploying flow: {}".format(repr(exc)), exc_info=exc)
```

Comment:
If using `exc_info`, you can drop the repr'ing of the exception in the log, since it's automatically included.

```suggestion
        self.logger.error("Error while deploying flow", exc_info=exc)
```

Reply:
Simplified the exception logging as suggested